src/core/router/identity.cc

```diff
@@ -435,60 +354,25 @@ SigningKeyType IdentityEx::GetSigningKeyType() const {
   if (m_StandardIdentity.certificate.type ==
       CERTIFICATE_TYPE_KEY && m_ExtendedBuffer)
     return bufbe16toh(m_ExtendedBuffer.get());  // signing key
-  return SIGNING_KEY_TYPE_DSA_SHA1;
+  // TODO(unassigned): should an exception be raised here?
+  // since an unsupported setup is used when the above condition is false?
+  std::unique_ptr<std::uint8_t[]> null_key = std::make_unique<std::uint8_t[]>(m_ExtendedLen);
+  return bufbe16toh(null_key.get());  // signing key
```

Comment:
>Likely not the correct solution, only fixes SIGILL runtime
interrupt.
>std::unique_ptr<std::uint8_t[]> null_key = std::make_unique<std::uint8_t[]>(m_ExtendedLen);
>return bufbe16toh(null_key.get());  // signing key

Not a real fix. If you drop this commit and rebase the branch, I can start sending in the patches I mentioned earlier.

Reply:
I can drop that commit no problem, where should I rebase the branch to? Just don't want all the comments to get deleted, unless that's not a problem.